src/vs/workbench/contrib/markers/browser/markersTreeViewer.ts

```diff
@@ -533,10 +533,16 @@ export class Filter implements ITreeFilter<MarkerElement, FilterData> {
 			return false;
 		}
 
-		const uriMatches = FilterOptions._filter(this.options.textFilter, basename(resourceMarkers.resource));
-
-		if (this.options.textFilter && uriMatches) {
-			return { visibility: true, data: { type: FilterDataType.ResourceMarkers, uriMatches } };
+		const negate = this.options.textFilter.startsWith('!');
+		const textFilter = this.options.textFilter.substring(negate ? 1 : 0);
```

Comment:
Seems like textFilter & negate options are being computed outside and everywhere. Instead I would suggest to move these semantics into FilterOptions and change `textFilter` type to `{text: string, negate: boolean} | undefined`.

Reply:
Good idea. I have pushed the changes.